plugins/auth-backend/src/providers/github/provider.ts

```diff
@@ -38,10 +38,60 @@ import {
 } from '../../lib/EnvironmentHandler';
 import { Logger } from 'winston';
 import { TokenIssuer } from '../../identity';
+import passport from 'passport';
 
 export class GithubAuthProvider implements OAuthProviderHandlers {
```

Comment:
To be able to act as a BackstageIdentity provider, GitHub auth needs implement the refresh endpoint as well, since the id tokens expire after 1h. It probably makes sense to just use the existing access token in a request to the refresh endpoint and fetch the profile again. It's a bit wonky though, and with a GitHub App you actually get a proper refresh flow with a refresh token, which is one of the reasons we hadn't implemented GitHub as a login provider yet.